Use Crashlytics NDK support

It seems that Crashlytics has NDK support: http://docs.fabric.io/android/crashlytics/ndk.html

Trying to copy the `build/android-arm-v7/Release/obj.target` directory contents to `android/java/MapboxGLAndroidSDK/src/main/obj` so we are compatible with `ndk-build` locations, but having trouble with the gyp.

So far this:


Gives me this error:
```
/bin/sh: 1: builddir: not found
ls: cannot access /obj.target/*: No such file or directory
gyp: Call to 'ls -1 $(builddir)/obj.target/*' returned exit status 2.
scripts/main.mk:83: recipe for target 'Makefile/__project__' failed
make[1]: *** [Makefile/__project__] Error 1
make[1]: Leaving directory '/home/leith/mb/mapbox-gl-native'
Makefile:71: recipe for target 'android-lib' failed
make: *** [android-lib] Error 2
```

I was trying to use the wild card trick from http://stackoverflow.com/a/32136063

I had tried this before:
```
        {
        'files': [
          '<(PRODUCT_DIR)/obj.target'
        ],
        'destination': '<(pwd)/../android/java/MapboxGLAndroidSDK/src/main/obj'
        },
```

But it copies the `obj.target` *inside* `obj` like: `android/java/MapboxGLAndroidSDK/src/main/obj/obj.target` instead of the contents.

@kkaefer Can you help me with ^

--- a/Makefile
+++ b/Makefile
@@ -131,7 +131,7 @@ clean: clear_sqlite_cache clear_xcode_cache
 	-rm -rf ./test/build
 	-rm -rf ./config/*.gypi
 	-rm -rf ./android/java/build ./android/java/MapboxGLAndroidSDK/build ./android/java/MapboxGLAndroidSDKTestApp/build
-	-rm -rf ./android/java/MapboxGLAndroidSDK/src/main/jniLibs ./android/java/MapboxGLAndroidSDK/src/main/assets
+	-rm -rf ./android/java/MapboxGLAndroidSDK/src/main/libs ./android/java/MapboxGLAndroidSDK/src/main/obj.target ./android/java/MapboxGLAndroidSDK/src/main/assets
 	-rm -f ./android/test/features.zip
 
 distclean: clean

--- a/android/java/.gitignore
+++ b/android/java/.gitignore
@@ -10,6 +10,12 @@ build/
 *.so
 *.apk
 
+# JNI
+MapboxGLAndroidSDK/src/main/libs/
+MapboxGLAndroidSDK/src/main/jniLibs/
+MapboxGLAndroidSDK/src/main/obj/
+MapboxGLAndroidSDK/src/main/obj.target/
+
 # Lib assets
 MapboxGLAndroidSDK/src/main/assets/
 

--- a/android/java/MapboxGLAndroidSDKTestApp/build.gradle
+++ b/android/java/MapboxGLAndroidSDKTestApp/build.gradle
@@ -80,6 +80,21 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+
+    splits {
+        abi {
+            enable true
+            reset()
+            include 'armeabi', 'armeabi-v7a', 'x86', 'mips'
+            universalApk true
+        }
+    }
+}
+
+crashlytics {
+    enableNdk true
+    androidNdkOut '../MapboxGLAndroidSDK/src/main/obj.target'
+    androidNdkLibsOut '../MapboxGLAndroidSDK/src/main/obj.target'
 }
 
 dependencies {
@@ -91,6 +106,9 @@ dependencies {
     compile 'com.android.support:appcompat-v7:23.0.1'
     compile 'com.android.support:design:23.0.1'
     compile('com.crashlytics.sdk.android:crashlytics:2.5.1@aar') {
+        transitive = true;
+    }
+    compile('com.crashlytics.sdk.android:crashlytics-ndk:1.1.1@aar') {
         transitive = true;
     }
 

--- a/android/java/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxgl/testapp/MainActivity.java
+++ b/android/java/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxgl/testapp/MainActivity.java
@@ -25,6 +25,7 @@ import android.view.View;
 import android.widget.TextView;
 import android.widget.Toast;
 import com.crashlytics.android.Crashlytics;
+import com.crashlytics.android.ndk.CrashlyticsNdk;
 import com.mapbox.mapboxgl.annotations.Marker;
 import com.mapbox.mapboxgl.annotations.MarkerOptions;
 import com.mapbox.mapboxgl.annotations.PolygonOptions;
@@ -78,7 +79,7 @@ public class MainActivity extends AppCompatActivity {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        Fabric.with(this, new Crashlytics());
+        Fabric.with(this, new Crashlytics(), new CrashlyticsNdk());
 
         // Load the layout
         setContentView(R.layout.activity_main);

--- a/android/mapboxgl-app.gypi
+++ b/android/mapboxgl-app.gypi
@@ -74,13 +74,19 @@
           ],
           'destination': '<(pwd)/../android/java/MapboxGLAndroidSDK/src/main/assets'
         },
+        {
+        'files': [
+          '<(PRODUCT_DIR)/obj.target'
+        ],
+        'destination': '<(pwd)/../android/java/MapboxGLAndroidSDK/src/main'
+        },
       ],
 
       'actions': [
         {
           'action_name': 'Strip dynamic library',
           'inputs': [ '<(PRODUCT_DIR)/lib.target/libmapbox-gl.so' ],
-          'outputs': [ '<(pwd)/../android/java/MapboxGLAndroidSDK/src/main/jniLibs/$(JNIDIR)/libmapbox-gl.so' ],
+          'outputs': [ '<(pwd)/../android/java/MapboxGLAndroidSDK/src/main/libs/$(JNIDIR)/libmapbox-gl.so' ],
           'action': [ '$(STRIP)', '<@(_inputs)', '-o', '<@(_outputs)' ]
         },
       ],

--- a/scripts/android/run.sh
+++ b/scripts/android/run.sh
@@ -39,16 +39,20 @@ if [ ! -z "${AWS_ACCESS_KEY_ID}" ] && [ ! -z "${AWS_SECRET_ACCESS_KEY}" ] ; then
 
     S3_PREFIX=s3://mapbox/mapbox-gl-native/android/build/${TRAVIS_JOB_NUMBER}
     APK_OUTPUTS=./android/java/MapboxGLAndroidSDKTestApp/build/outputs/apk
+    JNILIB=`mason env JNIDIR`
 
-    # Upload either the debug or the release build
-    if [ ${BUILDTYPE} == "Debug" ] ; then
-        aws s3 cp \
-            ${APK_OUTPUTS}/MapboxGLAndroidSDKTestApp-debug.apk \
-            ${S3_PREFIX}/MapboxGLAndroidSDKTestApp-debug.apk
-    elif [ ${BUILDTYPE} == "Release" ] ; then
-        aws s3 cp \
-            ${APK_OUTPUTS}/MapboxGLAndroidSDKTestApp-release-unsigned.apk \
-            ${S3_PREFIX}/MapboxGLAndroidSDKTestApp-release-unsigned.apk
+    # ARM64 does not build APK for now
+    if [ ${JNIDIR} != "arm64-v8a" ] ; then
+        # Upload either the debug or the release build
+        if [ ${BUILDTYPE} == "Debug" ] ; then
+            aws s3 cp \
+                ${APK_OUTPUTS}/MapboxGLAndroidSDKTestApp-${JNILIB}-debug.apk \
+                ${S3_PREFIX}/MapboxGLAndroidSDKTestApp-debug.apk
+        elif [ ${BUILDTYPE} == "Release" ] ; then
+            aws s3 cp \
+                ${APK_OUTPUTS}/MapboxGLAndroidSDKTestApp-${JNILIB}-release-unsigned.apk \
+                ${S3_PREFIX}/MapboxGLAndroidSDKTestApp-release-unsigned.apk
+        fi
     fi
 
     mapbox_time_finish

